client/material/button.tsx

```diff
@@ -272,12 +246,54 @@ export function useButtonState({
       onMouseDown: handleMouseDown,
       onMouseEnter: handleMouseEnter,
       onMouseLeave: handleMouseLeave,
+      onKeyDown: handleKeyDown,
+      onKeyUp: handleKeyUp,
       $focused: focused,
     },
     rippleRef,
   ]
 }
 
+export interface HotkeyProp {
+  keyCode: number
+  altKey?: boolean
+  shiftKey?: boolean
+  ctrlKey?: boolean
+}
+
+export interface ButtonHotkeyProps {
+  /** The reference to the button that should be pressed programmatically. */
+  ref?: React.MutableRefObject<HTMLButtonElement | undefined>
+  /** Whether the button is disabled (hotkey will do nothing). */
+  disabled?: boolean
+  /**
+   * A hotkey to register for the button. Pressing the specified modifiers and key will result in
+   * the button being clicked programmatically.
+   */
+  hotkey?: HotkeyProp
```

Comment:
I don't really get why this is optional either, what is the point of using this if you don't have a hotkey?

Reply:
Okay I looked lower and I see now. This is not a good structure, this makes it so every single button will register a global key listener, which is potentially expanding the number of key listeners drastically. `useButtonHotkey` operates only on a ref, I don't see why the button types need any understanding of hotkeys whatsoever once you've added this hook. So you should make these fields required and remove the usage from the button types below.